packages/mui-joy/src/IconButton/IconButtonProps.ts

```diff
@@ -16,12 +16,18 @@ export interface IconButtonSlots {
    * @default 'button'
    */
   root?: React.ElementType;
+  /**
+   * The component that renders the loading indicator.
+   * @default 'span'
+   */
+  loadingIndicator?: React.ElementType;
```

Comment:
Need to update the `IconButtonSlot` type above as well:
```ts
export type IconButtonSlot = 'root' | 'loadingIndicator';
```

Additionally, could you please add a type test [here](https://github.com/mui/material-ui/blob/master/packages/mui-joy/src/styles/extendTheme.spec.ts#L699) for `loadingIndicator` slot?